### PR TITLE
alluxio 2.4.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
==> Testing alluxio
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201122-15950-5vfaex.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/alluxio.rb --verbose
==> /usr/local/Cellar/alluxio/2.4.1/bin/alluxio version
Error: Alluxio requires Java 8 or Java 11, currently Java  found.
Error: alluxio: failed
An exception occurred within a child process:
  BuildError: Failed executing: /usr/local/Cellar/alluxio/2.4.1/bin/alluxio version
```

File: Formula/alluxio.rb [hunk 1]
```diff
@@ -20,6 +24,7 @@ def default_alluxio_conf
   def install
     doc.install Dir["docs/*"]
     libexec.install Dir["*"]
+    bin.env_script_all_files libexec/"bin", JAVA_HOME: Formula["openjdk@11"].opt_prefix
```
Comment (SMillerDev) [inline Formula/alluxio.rb:27]: Why not allow this to be overridable?

File: Formula/alluxio.rb [hunk 2]
```diff
@@ -41,6 +46,6 @@ def caveats
   end
 
   test do
-    system bin/"alluxio", "version"
+    assert_match version.to_s, shell_output("#{bin}/alluxio version")
```
Comment (SMillerDev) [inline Formula/alluxio.rb:49]: We should get a better test here.

File: Formula/alluxio.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/alluxio.rb:27]: yeah, I was not very sure both the syntax.

But yeah, I should do that.

File: Formula/alluxio.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/alluxio.rb:49]: totally!

**Comment by chenrui333** [comment]
Still failed with the java pickup issue on CI.
```
==> /usr/local/Cellar/alluxio/2.4.1/bin/alluxio version
Error: Alluxio requires Java 8 or Java 11, currently Java  found.
Error: alluxio: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<1>.
```

**Comment by SeekingMeaning** [comment]
Fails on Big Sur ARM since it doesn't have a bottle for `openjdk@11`; merging

**Pull request merged**